__tests__/V5ClusterManagement.js

```diff
@@ -0,0 +1,458 @@
+  expect(inputMin.value).toBe(defaultScaling.min.toString());
+  expect(inputMax.value).toBe(defaultScaling.max.toString());
+
+  // Change the values and modify the scaling settingsw
+  fireEvent.change(inputMin, { target: { value: newScaling.min } });
+  fireEvent.change(inputMax, { target: { value: newScaling.max } });
+  const textButton = `Increase minimum number of nodes by ${newScaling.min}`;
+
+  await wait(() => getByText(textButton));
+
+  const submitButton = getByText(textButton);
+  fireEvent.click(submitButton);
+
+  //Wait for the Flash message to appear
+  await wait(() => {
+    getByText(/succesfully edited node pool name/i);
+  });
+
+  // Our node pool is the first one. Does it have the scaling values updated?
+  expect(getAllByTestId('scaling-min')[0]).toHaveTextContent(
+    newScaling.min.toString()
+  );
+  expect(getAllByTestId('scaling-max')[0]).toHaveTextContent(
+    newScaling.max.toString()
+  );
+
+  nodePoolPatchRequest.done();
+});
+
+it('deletes a v5 cluster', async () => {
+  const cluster = v5ClusterResponse;
+  const clusterDeleteResponse = {
+    code: 'RESOURCE_DELETION_STARTED',
+    message: `Deletion of cluster with ID '${V5_CLUSTER.id}' is in progress.`,
+  };
+
+  // Request
+  const clusterDeleteRequest = nock(API_ENDPOINT)
+    .intercept(`/v4/clusters/${V5_CLUSTER.id}/`, 'DELETE')
+    .reply(200, clusterDeleteResponse);
+
+  const div = document.createElement('div');
+  const {
+    getByText,
+    getAllByText,
+    queryByTestId,
+    getAllByTestId,
+    debug,
+    container,
+  } = renderRouteWithStore(ROUTE, div, {});
+
+  // Wait for the view to render
+  await wait(() => {
+    expect(getByText(V5_CLUSTER.name)).toBeInTheDocument();
+  });
+
+  // fireEvent.click(getAllByText('•••')[0]);
+  await wait(() => getByText('Delete Cluster'));
+  fireEvent.click(getByText('Delete Cluster'));
+
+  // Is the modal in the document?
+  const titleText = /are you sure you want to delete/i;
+  await wait(() => getByText(titleText));
+  const modalTitle = getByText(titleText);
+  expect(modalTitle).toBeInTheDocument();
+  expect(modalTitle.textContent.includes(cluster.id)).toBeTruthy();
+
+  // Click delete button.
+  const modalDeleteButton = getAllByText('Delete Cluster')[1];
+  fireEvent.click(modalDeleteButton);
+
+  // Flash message confirming deletion.
+  await wait(() => {
+    getByText(/will be deleted/i);
+  });
+  const flashElement = getByText(/will be deleted/i);
+  expect(flashElement).toBeInTheDocument();
+  expect(flashElement).toHaveTextContent(cluster.id);
+
+  // Expect the cluster is not in the clusters list.
+  await wait(() => {
+    expect(queryByTestId(cluster.id)).not.toBeInTheDocument();
+  });
+
+  // This is not inside the component tree we are testing and so it is not cleaned up
+  // after test, so we have to remnove it manually in order to not cause conflicts with
+  // the next test with a flash message
+  document.querySelector('#noty_layout__topRight').remove();
```

Comment:
This feels a bit weird to me. The tests share a "document " somehow? I didn't know that.

Reply:
Yeah we are testing with a simulator of athe browser: `jsdom`

This is regarding what I wrote above: https://github.com/giantswarm/happa/pull/832#discussion_r347935946

Reply:
yep,  `jest` uses `jsdom` wich is a browser simulator and produces a document object